src/hooks/useConfirm.ts

```diff
@@ -0,0 +1,29 @@
+import { useContext } from 'react';
+import { ConfirmContext as context } from '../context/ConfirmContext';
+
+const useConfirm = () => {
+  const { message, resolve, setMessage, setResolve } = useContext(context);
+
+  const isOpen = !!message;
+
+  const confirm = (message: string) => {
+    setMessage(message);
+    return new Promise<boolean>((resolve) => {
+      setResolve(() => resolve);
+    });
+  };
+
+  const onConfirm = () => {
+    resolve?.(true);
+    setMessage('');
+  };
+
+  const onCancel = () => {
+    resolve?.(false);
+    setMessage('');
+  };
+
+  return { message, isOpen, confirm, onConfirm, onCancel };
```

Comment:
상단에서 message로 관리되고 있고 `isOpen`은 여러 로직상 따로 쓰이고 있진 않아서  다음과 같이 지정해도 되지 않을까요?

```suggestion
  return { message, isOpen:!!message, confirm, onConfirm, onCancel };
```

Reply:
아, 이건 제 스타일이기도 한데요. return 부분에선 return만 하고 값을 할당하지 않는 것을 선호하는 편입니다😅

추가로 isOpen이 따로 쓰이고 있지 않다는 말씀이 이해가 잘 가지 않는데 좀 더 자세히 설명 부탁드려도 될까요?

Reply:
isOpen이 상단에 정의만 되고 다른 함수(conFirm, onConfirm, onCancel)에서 따로 쓰이지 않고 있어서, return할 때 바로 넘겨줘도 되지않을까 해서 제안을 드린것이었습니다!

Reply:
[commit] [87cd073](https://github.com/frontend-opensource-project/use-react-hooks/pull/16/commits/87cd07372be255eaad6e4acc6fe20b41f25f2216)

@foresec 선언부를 따로 분리하는 게 좋다고 생각했었는데 훅이 복잡하지 않은 상태에선 반환부에 직접 정의하는 것도 괜찮겠다고 생각해서 반영했습니다! 확인 후 Approve 부탁드려요!